config/initializers/console.rb

```diff
@@ -10,4 +10,32 @@ def find(id)
       raise "Don't know how to resolve this ¯\\_(ツ)_/¯. Please provide a valid email or Global ID."
     end
   end
+
+  # If there are credit notes, the function will fail.
+  def hard_delete_invoice(id)
+    invoice = Invoice.find(id)
+    puts "Going to hard delete invoice from org `#{invoice.organization.name}` (id: #{invoice.id})"
+
+    puts "Press any key to confirm deletion or CTRL+C to cancel."
+    c = STDIN.getch
+
+    if c == "\u0003"
+      puts "Deletion cancelled."
+      return invoice
+    end
+
+    puts "Deleting invoice #{invoice.id}..."
+    ActiveRecord::Base.transaction do
```

Comment:
I think you need to also remove few other associated objects:
```
invoice.invoice_subscriptions.destroy_all
invoice.credit_notes.destroy_all
invoice.fees.each { |f| f.true_up_fee&.destroy! }
invoice.fees.destroy_all
invoice.taxes.destroy_all
invoice.credits.destroy_all
invoice.destroy
```

Reply:
Thank you @lovrocolic! I updated the code.
I wasn't too sure about deleting invoices with credit notes but I'm sure when we use this helper, we know what we're doing. 🙏